pkg/kubectl/scale.go

```diff
@@ -71,11 +71,23 @@ func (pe PreconditionError) Error() string {
 
 // RetryParams encapsulates the retry parameters used by kubectl's scaler.
 type RetryParams struct {
-	Interval, Timeout time.Duration
+	Interval    time.Duration
+	Timeout     time.Duration
+	RetryErrors bool
```

Comment:
I don't think that's the right way of doing that.
This RetryError is something won't be used in production code. So effectively what you're doing here is you're polluting production code with testing code. That's usually bad decision.

This should rather use buliding block from scaler and probably duplicate some logic into test/utils/update.go

In particular, without touching kubectl code at all, in test/utils/update.go you can do sth like:
```
cond := ScaleCondition(...)
// opaque that into RetryErrorCondition.
if err := wait.PollImediate()... {}
return wait.PollImeddiate(.., /*here will be your (probably a bit custom) version of scaleHasDesiredReplicas)

Reply:
OK, I understand.
I will try different approach then.